### PR TITLE
Test: Explorer View

### LINKED ISSUE #635: Service Instance View: unit test: Explorer view

### PR DESCRIPTION
**Pull request by kaitmore**
Resolves #635 

## Proposed Changes

  - Tests for Explorer and Inspector components including snapshots
  
## How to Test

  - `npm test`

### CODE REVIEW

File: src/components/Main/components/Explorer/components/Inspector.test.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  "http/connections",
+  "http/connects"
+];
+
+const mockProps = {
+  data: mockData,
+  onClick: jest.fn(),
+  onSearch: jest.fn(),
+  searchQuery: "",
+  selectedMetric: undefined
```
Comment (tilleryd) [inline src/components/Main/components/Explorer/components/Inspector.test.js]: Setting a value to `undefined` is the same as not defining it i.e. you could just not define it as a key/value pair of the object.  If you want to be explicit about it's presence but set it to some value then `null` would suffice.

**Comment by kaitmore** [comment]
@tilleryd I addressed your comment 😀

File: src/components/Main/components/Explorer/explorer.test.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  });
+
+  test("passes the correct props down to Inspector", () => {
+    const instance = wrapper.find("Explorer").instance();
+    const expectedProps = {
+      data: instance.props.keys,
+      onClick: instance.onClick,
+      onSearch: instance.onSearch,
+      searchQuery: "",
+      selectedMetric: undefined,
```
Comment (tilleryd) [inline src/components/Main/components/Explorer/explorer.test.js]: Set this to `null` as well.

File: src/components/Main/components/Explorer/explorer.test.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  beforeEach(() => {
+    instance = mount(RouterWrap)
+      .find("Explorer")
+      .instance();
+  });
+  test("_pushHistory sets state and pushes queryString to browser history", () => {
+    instance._pushHistory("jvm");
+
+    expect(instance.state.lastPushedQueryString).toBe("jvm");
+    expect(instance.props.history.location.search).not.toBe(null);
```
Comment (tilleryd) [inline src/components/Main/components/Explorer/explorer.test.js]: If the next test is to test that `history.location.search` is `?jvm` then I think the test for null is moot and unneeded.

File: src/components/Main/components/Explorer/explorer.test.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    expect(instance.props.history.location.search).not.toBe("");
+    instance._appendToQueryString({ selectedMetric: "RPC%2Frequests" });
+    expect(instance.props.history.location.search).toBe(
+      "?selectedMetric=RPC%252Frequests"
+    );
+    expect(instance.props.history.location.search).not.toBe(
+      "?selectedMetric=HTTP%252Frequests"
+    );
+  });
+});
```
Comment (tilleryd) [inline src/components/Main/components/Explorer/explorer.test.js]: I hate to be a stickler but can you separate each test with a new line?

File: src/components/Main/components/Explorer/explorer.test.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    expect(instance.state.lastPushedQueryString).toBe("jvm");
+    expect(instance.props.history.location.search).not.toBe(null);
+    expect(instance.props.history.location.search).toBe("?jvm");
+    expect(instance.props.history).toHaveLength(2);
+  });
+
+  test("popAndDecodeHistory parses the query string and sets local state", () => {
+    instance.popAndDecodeHistory("?searchQuery=deviation");
+    expect(instance.state.searchQuery).toBe("deviation");
+    expect(instance.state.searchQuery).not.toBe("");
```
Comment (tilleryd) [inline src/components/Main/components/Explorer/explorer.test.js]: This assertion is moot given the test above it right?

File: src/components/Main/components/Explorer/explorer.test.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+  });
+
+  test("onSearch sets local state with user's search query", () => {
+    instance.onSearch("status");
+    expect(instance.state.searchQuery).toBe("status");
+    expect(instance.state.searchQuery).not.toBe("");
```
Comment (tilleryd) [inline src/components/Main/components/Explorer/explorer.test.js]: This assertion is moot given the assertion above it right?

File: src/components/Main/components/Explorer/explorer.test.js [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+  });
+  test("onClick escapes slashes and calls _appendToQueryString", () => {
+    instance.onClick("HTTP/requests");
+    expect(instance.props.history.location.search).toBe(
+      "?selectedMetric=HTTP%252Frequests"
+    );
+    expect(instance.props.history.location.search).not.toBe("");
```
Comment (tilleryd) [inline src/components/Main/components/Explorer/explorer.test.js]: This assertion is moot given the assertion above it right?

**Comment by kaitmore** [comment]
@tilleryd Addressed your comments 👍🏼

**Pull request merged**